src/Propel/Runtime/ActiveQuery/InstancePoolTrait.php

```diff
@@ -30,12 +30,14 @@ public static function addInstanceToPool($object, $key = null)
     public static function getInstanceKey($value)
     {
         if (!($value instanceof Criteria) && is_object($value)) {
-            if (count($pk = $value->getPrimaryKey()) > 1 || is_object($value->getPrimaryKey())) {
+            $pk = $value->getPrimaryKey();
```

Comment:
Sorry, I'm just snooping around. Should:

    (is_numeric($pk) && $pk > 1)

be

    (is_numeric($pk) && count($pk) > 1)

Reply:
@denormalizer - a numeric value won't be "countable" which is the source of the issue; since we know it is numeric on the conditional it would just be 

```(is_numeric($pk) && $pk > 1)```

Reply:
True. I don't know what I was thinking :)

Reply:
Thanks that worked well!